### PR TITLE
trivial: fix examples explaining IRQT_TO_IDX macro

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
This PR just fixes examples given in a comment for how an IRQ gets converted to a CNode index.

Example 2: The IRQ number is not a PPI so the index is `(irq.target_core)*NUM_PPI + (_irq.irq)`, which means it should be `+ 16` not `+ 12`.

Example 3: The IRQ number is a PPI so the index is `(CONFIG_MAX_NUM_NODES-1)*NUM_PPI + (_irq.irq)`, which means it should be `(3 * 32) + 33` since there's 4 cores, not sure where the `(33 - 32)` came from.

### CODE REVIEW

File: include/arch/arm/arch/machine/gic_common.h [hunk 1]
```diff
@@ -37,8 +37,8 @@
  * The layout of IRQs into the CNode are all of PPI's for each core first, followed
  * by the global interrupts.  Examples:
  *   core: 0, irq: 12 -> index 12.
- *   core: 2, irq: 16 -> (2 * 32) + 12
- *   core: 1, irq: 33, (4 total cores) -> (4 * 32) + (33-32).
+ *   core: 2, irq: 16 -> (2 * 32) + 16
```
Comment (axel-h) [inline include/arch/arm/arch/machine/gic_common.h:40]: You are right, this should be 16. Seems it's a copy/paste bug, form above.

File: include/arch/arm/arch/machine/gic_common.h [hunk 2]
```diff
@@ -37,8 +37,8 @@
...
+ *   core: 1, irq: 33, (4 total cores) -> (3 * 32) + 33.
```
Comment (yyshen) [inline include/arch/arm/arch/machine/gic_common.h:41]: The previous example is clear to me. 4 * 32 clearly expresses that the first 4*32 slots are for the PPI and SGI of the 4 cores.

Comment (axel-h) [inline include/arch/arm/arch/machine/gic_common.h:41]: Sees the `(33-32)` comes from seeing it as `(CONFIG_MAX_NUM_NODES * NUM_PPI + (_irq.irq - NUM_PPI))`, but at some point this was simplified to `(CONFIG_MAX_NUM_NODES - 1) * NUM_PPI + (_irq.irq)` then in the code.

**Comment by Ivan-Velickovic** [comment]
The third example is correct either way, so I've decided to leave it and just fix the mistake in the second example.

File: include/arch/arm/arch/machine/gic_common.h (hunk 2 above)
Comment (kent-mcleod) [reply include/arch/arm/arch/machine/gic_common.h]: Axel's right that the implementation was simplified.  The example is trying to explain the design and should be illustrating that the SPI interrupts start from 0 at the end of the PPI/SGI of all the cores.

**Review by kent-mcleod — APPROVED** [review]

**Review by lsf37 — APPROVED** [review]

**Pull request merged**